test/support/route.go

```diff
@@ -36,3 +41,57 @@ func GetRoute(t Test, namespace, name string) *routev1.Route {
 	t.T().Helper()
 	return Route(t, namespace, name)(t)
 }
+
+func ExposeServiceByRoute(t Test, name string, namespace string, serviceName string, servicePort string) url.URL {
+	r := &routev1.Route{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: routev1.SchemeGroupVersion.String(),
+			Kind:       "Route",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: routev1.RouteSpec{
+			To: routev1.RouteTargetReference{
+				Name: serviceName,
+			},
+			Port: &routev1.RoutePort{
+				TargetPort: intstr.FromString(servicePort),
+			},
+		},
+	}
+
+	_, err := t.Client().Route().RouteV1().Routes(r.Namespace).Create(t.Ctx(), r, metav1.CreateOptions{})
+	t.Expect(err).NotTo(gomega.HaveOccurred())
+	t.T().Logf("Created Route %s/%s successfully", r.Namespace, r.Name)
+
+	t.T().Logf("Waiting for Route %s/%s to be available", r.Namespace, r.Name)
+	t.Eventually(Route(t, r.Namespace, r.Name), TestTimeoutLong).
+		Should(gomega.WithTransform(ConditionStatus(routev1.RouteAdmitted), gomega.Equal(corev1.ConditionTrue)))
+
+	// Retrieve hostname
+	r, err = t.Client().Route().RouteV1().Routes(r.Namespace).Get(t.Ctx(), r.Name, metav1.GetOptions{})
+	t.Expect(err).NotTo(gomega.HaveOccurred())
+	hostname := r.Status.Ingress[0].Host
+
+	// Wait for expected HTTP code
+	t.Eventually(func() int {
+		resp, _ := http.Get("http://" + hostname)
+		return resp.StatusCode
+	}, TestTimeoutLong).Should(gomega.Not(gomega.Equal(503)))
+
+	r = GetRoute(t, r.Namespace, r.Name)
+
+	scheme := "http"
+	if r.Spec.Port.TargetPort.String() == "https" {
+		scheme = "https"
+	}
```

Comment:
When I think about it again, this is not actually needed, as above we construct HTTP route anyway, so we are sure it is HTTP.
Can you please remove this code?